compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -811,44 +820,142 @@ static void buildRecordComparisonFunc(AggregateType* ct, const char* op) {
+        Expr* right = new CallExpr(tmp->name, gMethodToken, arg2);
+        CallExpr *elemCompTrue = new CallExpr("<", left, right);
+        fn->insertAtTail(new CondStmt(elemCompTrue,
                                       new CallExpr(PRIM_RETURN, gTrue)));
       }
-      else {
-        fn->insertAtTail(new CondStmt(new CallExpr("!", elemComp), 
+      {
+        Expr* left = new CallExpr(tmp->name, gMethodToken, arg1);
+        Expr* right = new CallExpr(tmp->name, gMethodToken, arg2);
+        CallExpr *elemCompFalse = new CallExpr(">", left, right);
+        fn->insertAtTail(new CondStmt(elemCompFalse,
                                       new CallExpr(PRIM_RETURN, gFalse)));
       }
     }
   }
+  // either it is an empty record, and we didn't add any meaningful comparisons,
+  // or we added those comparisons but they never returned
 
-  // find the return value and add the return statement
-  VarSymbol* ret;
-  if (fn->body->length() > 0) { // we added comparison statements: not empty
-    if (isNotEqual)
-      ret = gFalse;
-    else
-      ret = gTrue;
+  // for <, in either case we return false
+  if (allowEquals) {
+    fn->insertAtTail(new CallExpr(PRIM_RETURN, gTrue));
   }
-  else { // this is an empty record, so instances are always equal to each other
-    if (astrOp == astrSeq || astrOp == astrSlte || astrOp == astrSgte)
-      ret = gTrue;
-    else
-      ret = gFalse;
+  else {
+    fn->insertAtTail(new CallExpr(PRIM_RETURN, gFalse));
   }
-  fn->insertAtTail(new CallExpr(PRIM_RETURN, ret));
+}
 
-  DefExpr* def = new DefExpr(fn);
-  ct->symbol->defPoint->insertBefore(def);
-  reset_ast_loc(def, ct->symbol);
-  normalize(fn);
+static void buildRecordGreaterThanBody(AggregateType *ct, FnSymbol *fn,
+                                       ArgSymbol *arg1, ArgSymbol *arg2,
+                                       bool allowEquals) {
+  // add comparisons for fields
+  for_fields(tmp, ct) {
+    if (!tmp->hasFlag(FLAG_IMPLICIT_ALIAS_FIELD) &&
+        !tmp->hasFlag(FLAG_TYPE_VARIABLE)) {  // types fields must be equal
+      {
+        Expr* left = new CallExpr(tmp->name, gMethodToken, arg1);
+        Expr* right = new CallExpr(tmp->name, gMethodToken, arg2);
+        CallExpr *elemCompTrue = new CallExpr(">", left, right);
+        fn->insertAtTail(new CondStmt(elemCompTrue,
+                                      new CallExpr(PRIM_RETURN, gTrue)));
+      }
+      {
+        Expr* left = new CallExpr(tmp->name, gMethodToken, arg1);
+        Expr* right = new CallExpr(tmp->name, gMethodToken, arg2);
+        CallExpr *elemCompFalse = new CallExpr("<", left, right);
+        fn->insertAtTail(new CondStmt(elemCompFalse,
+                                      new CallExpr(PRIM_RETURN, gFalse)));
+      }
+    }
+  }
+  // either it is an empty record, and we didn't add any meaningful comparisons,
+  // or we added those comparisons but they never returned
+
+  // for <, in either case we return false
```

Comment:
Ditto here.  Also, both this and the previous function refer to `<`, but I think one of them should refer to `>` (it should be this one if I'm understanding correctly)